### PR TITLE
Inital draft for bluechi-is-online tests

### PR DESCRIPTION
**Pull request by DanaOrr**
Inital draft for bluechi-is-online tests
We need assistance with calling for dnf install of bluechi-is-online rpm.

### CODE REVIEW

File: tests/bluechi_machine_lib/bluechi_is_online.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+class BluechiIsOnline:
+    """
+    A Python module for the 'bluechi-is-online' command to check the status of agents, nodes, and the system.
+    """
+
+    def __init__(self, executable_path: str = "/usr/bin/bluechi-is-online"):
+        self.executable_path = executable_path
+
+    def _run_command(self, args: list) -> Tuple[int, str, str]:
```
Comment (mkemel) [inline tests/bluechi_machine_lib/bluechi_is_online.py:19]: I would align this class with `BluechiCtl` class in `bluechi_machine_lib`, maybe even share the `run()` code with inheritance or something.

File: tests/bluechi_machine_lib/bluechi_is_online.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            )
+
+            return process.returncode, stdout.strip(), stderr.strip()
+
+        except Exception as e:
+            print(f"Error while executing command '{command}': {e}")
+            raise
+
+
+class TestBluechiIsOnline(unittest.TestCase):
```
Comment (mkemel) [inline tests/bluechi_machine_lib/bluechi_is_online.py:40]: Note that this shouldn't be a test, but just a library to reuse when running code in the bluechi container/vm.

**Review by mkemel — COMMENTED** [review]
Some comments.
I know that I have referred you to `bluechi_machine_lib` code earlier, but on the second thought it might not be necessary in this specific case. 
Take a look also at `bluechi_test/bluechictl.py`. There is a difference between these two. 

Stuff in `bluechi_test/bluechictl.py` runs single command via Podman REST API/SSH to VM (let's call it a machine), while code in `bluechi_machine_lib` exists to be included by scripts that are copied to the machine. The former is easy and can be executed directly from the test, while the latter is more complex, but allows greater flexibility. You can look at the usage of both in different tests.

**Comment by mkemel** [comment]
On another look, `bluechi_machine_lib` would be needed to test `--wait` and `--monitor`

File: tests/tests/tier0/bluechi-is-online-agent/main.fmf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+summary: Test if bluechi agent is online
```
Comment (mwperina) [inline tests/tests/tier0/bluechi-is-online-agent/main.fmf:1]: Please make it clear that the test uses `bluechi-is-online` tool with `agent` parameter, for example:

```
summary: Test if bluechi-agent is online using `bluechi-is-online agent` command line tool
```

**Review by mwperina — CHANGES_REQUESTED** [review]
Hmm, not entirely how it should look like, here are some comments:

1. Looking at https://github.com/eclipse-bluechi/bluechi/issues/981 the test will require setup with a controller and a node, but here the setup is completely missing. As an example a simple setup with a node and a controller can be found at [test_bluechi_node_status](https://github.com/eclipse-bluechi/bluechi/blob/main/tests/tests/tier0/bluechi-node-status/test_bluechi_node_status.py#L43) method

2. The scenario around `bluechi-is-online [agent|node]` requires that `bluechi-is-online` is executed on either controller or node machine, so you should rather take a look https://github.com/eclipse-bluechi/bluechi/blob/main/tests/bluechi_test/bluechictl.py Please do not look at `bluechi_machine_lib`, this is more advanced use case not relevant to these test cases

3. Please use descriptive commit message, for example `Add test for the plain bluechi-is-online agent invocation`

4. Describe more clearly the content of the commit and reference the issue the commit is related to, for example:
   """
   - Adds bc_is_online.py module to simplify usage of bluechi-is-online tool from integration test code
   - Adds integration test for bluechi-is-online agent execution without additional parameters
   
   Fixes: https://github.com/eclipse-bluechi/bluechi/issues/981
   """
   
   When creating the PR either use the commit title/description for PR title/description (if PR contains just a single commit) or fill-in reasonable title and description with list of all commits (and again please don't forget to reference the issue)

5. When creating a commit, please sign it using `-s, --signoff` parameter and don't forget to use the same email for sign off which you use for registration of your Eclipse account. And also please make sure that the sign off is included in PR description

**Comment by engelmi** [comment]
I think this is a duplicate of https://github.com/eclipse-bluechi/bluechi/pull/996/ (which is already merged). Closing. 
If not, please reopen it.

**Pull request closed without merging**